odoo/addons/test_orm/tests/test_guess_mimetypes.py

```diff
@@ -1,115 +1,226 @@
 import os.path
+import typing
 import unittest
 
 from odoo.tests import tagged, BaseCase
-from odoo.tools.mimetypes import _odoo_guess_mimetype, guess_mimetype
-from odoo.tools.misc import file_open
+from odoo.tools.mimetypes import _odoo_guess_mimetype, _odoo_guess_file_mimetype, fix_filename_extension, get_extension, guess_mimetype, guess_file_mimetype
 
 
-def contents(extension):
-    with file_open(os.path.join(
-        os.path.dirname(__file__),
-        'testfiles',
-        f'case.{extension}'
-    ), 'rb') as f:
-        return f.read()
+TEST_FOLDER = os.path.join(os.path.dirname(__file__), 'testfiles')
 
 
 class MimeGuessingCases:
-    allow_inherited_tests_method = True
-    guess_mimetype: callable
+    allow_inherited_tests_method: bool = True
+    guess_mimetype: typing.Callable[[bytes, str], str]
+    guess_file_mimetype: typing.Callable[[str, str], str]
 
     def test_doc(self):
         self.assertEqual(
-            self.guess_mimetype(contents('doc')),
+            self.guess_file_mimetype(f'{TEST_FOLDER}/case.doc'),
```

Comment:
If `TEST_FOLDER` is changed for a `Path`
```suggestion
            self.guess_file_mimetype(TEST_FOLDER / 'case.doc'),
```